mne/preprocessing/tests/test_ecg.py

```diff
@@ -79,4 +79,18 @@ def test_find_ecg():
     assert 'ECG-SYN' not in ecg_epochs.ch_names
 
 
+def test_find_ecg_events_tstart():
+    """Ensure tstart is taken into account when calculating avg heart rate."""
+    raw = read_raw_fif(raw_fname, preload=False)
+    event_id = 999
+    tstart = raw.times[-1] / 2
+    events, _, average_hr = find_ecg_events(raw=raw, event_id=event_id,
+                                            tstart=tstart)
+
+    duration_in_sec = len(raw) / raw.info['sfreq'] - tstart
+    duration_in_min = duration_in_sec / 60
+    average_hr_expected = len(events) / duration_in_min
+    assert_allclose(average_hr, average_hr_expected)
```

Comment:
can you rather update an existing test or maybe use pytest factorize with tstart value?

Reply:
Didn't want to touch the (one) existing test because it seems to be mostly concerned with testing proper function on not-preloaded data (and then some bits I don't understand at all); I've now added pytest parametrization. Test function takes about 0.75 sec on my machine now

Reply:
I suggested this to reduce computation time of tests or test this all at the same place.

it turns out it's tested here 

https://github.com/mne-tools/mne-python/blob/971f631487975fa0108c86d8ce9203afc2def6b6/mne/preprocessing/tests/test_ecg.py#L41

can we factorize this to try? objective: keep testing time low and centralize tests in one place. thanks !

Reply:
Ok I have addressed this in 7477f68
Time for running the test function has not changed for me compared to `master`! (ignoring between-run variability)

Reply:
Slightly improved test coverage via ee58ebb, we're now ~150 ms slower than on `master`. I think this is okay?